slang-com-ptr.h

```diff
@@ -1,16 +1,15 @@
-#ifndef SLANG_COM_PTR_H
+﻿#ifndef SLANG_COM_PTR_H
 #define SLANG_COM_PTR_H
 
-#include "slang-defines.h"
-#include "slang-result.h"
+#include "slang-com-helper.h"
 
 #include <assert.h>
 
 namespace Slang {
```

Comment:
The same namespace thing arises here as before: the current convention would be toward this being `slang` for public API, rather than `Slang`, but I'm fine with entertaining a switch.

Reply:
I'm not sure what the right answer is. If we want an internal slang namespace - we should probably just name it 'slang_internal'/'slangInternal'/'SlangInternal'/'Slang::Internal' - or whatever convention we want to use. 

There is a school of thought that says namespaces should be lower snake - this seems to be 'inspired' by STL. There is another school of thought that says they should be upper camel - that seems to be inspired more by languages like Java/C#. 

Less common seem to be lower camel, or just lower case (!). I think we can rule out just lower camel cos it seems uncommon and less readable. Lower camel again seems unusual - but at least it's more readable than just lower case.

That leaves lower snake and upper camel. The 'advantage' seen with lower camel (other than STL) is that it makes it clear that it's a namespace. The 'advantage' of upper camel (other than similarity to Java/C#) is sort of the opposite - that it sort of unifies them, as typically you don't want or need to know the difference. 

So lets call that a tie. Perhaps the tie breaker is what kind of API are you producing. Does it have methods with lower camel (like STL)? Or lower camel (like C#/Java)? Then perhaps it can be argued your name spacing convention should follow the conventions of the thing that you are closest too.

By this argument upper camel matches most closely slangs other naming conventions.  That said - this isn't a killer argument, and if you wanted to make it clearer where a namespace was being used lower camel, would be acceptable. 

I would also lay bare my preference is where all things being equal for upper camel - for readability, conciseness and continuity reasons.